packages/kit/src/runtime/client/router.js

```diff
@@ -448,6 +450,10 @@ export class Router {
 		}
 
 		if (details) {
+			if (this.is_fallback && info.url.pathname !== location.pathname) {
```

Comment:
we should probably add a comment to clarify at least
```suggestion
			// if navigating to a prerendered page
			if (this.is_fallback && info.url.pathname !== location.pathname) {
```

but another way to write this might be to move the URL check to `renderer`. overwriting `is_fallback` seems a bit funny to me because at this point it is still a fallback. it's not until after we've left and navigated to the new page that it's not a fallback

Reply:
Not sure I understand what you mean about moving the URL check to `renderer`. The responsibility for this stuff (to the extent that the two things are separate — I'm going to open a PR today to merge them, because it really will simplify things) lies with the router

> it's not until after we've left and navigated to the new page that it's not a fallback

pushing state _is_ navigating to the new page

Reply:
I mean you could change `if (!navigation_result && this.router?.is_fallback) {` to `if (!navigation_result && this.router?.is_fallback && info.url.pathname !== location.pathname) {`

If I understand correctly, when this is triggered we will do a server-side nav. This page is still a fallback page in my mind until the server-side nav has happened

Reply:
Actually you know what? If we were to compare `url.pathname` with `location.pathname` at that point, it would arguably be simpler to do away with the fallback/non-fallback distinction altogether and just always bail out of full page reloads if they're the same (i.e. it's not a navigation, it's a hydration/invalidation). commit inbound

Reply:
Yeah, that's simpler and should work.